WeatherApp/ViewControllers/SearchCityViewController.swift

```diff
@@ -0,0 +1,77 @@
+import UIKit
+
+protocol SearchCityDelegate: AnyObject {
+    func passCoordinate(latitude: Double, longitude: Double)
+}
+
+class SearchCityViewController: UIViewController {
+    
+    @IBOutlet weak var searchCity: UISearchBar!
+    @IBOutlet weak var locationTableView: UITableView!
+    
+    var locationList: [CityModel]?
+    var searchData = [CityModel]()
+    weak var delegate: SearchCityDelegate?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        prepareCityList()
+        registerCustomLocationCell()
+        searchData = locationList ?? []
+    }
+    
+    func registerCustomLocationCell() {
+        let nib = UINib(nibName: "LocationTableViewCell", bundle: nil)
+        locationTableView.register(nib, forCellReuseIdentifier: "locationTableViewCell")
+    }
+    
+    func prepareCityList() {
+        let dhaka = CityModel(name: "Dhaka",latitude: 23.7104,longitude: 90.4074)
+        let rajshahi = CityModel(name: "Rajshahi",latitude: 24.3667,longitude: 88.6)
+        let khulna = CityModel(name: "Khulna",latitude: 22.8135,longitude: 89.5672)
+        let rangpur = CityModel(name: "Rangpur",latitude: 25.75,longitude: 89.25)
+        let barishal = CityModel(name: "Barishal",latitude: 22.5,longitude: 90.3333)
+        let chattogram = CityModel(name: "Chattogram",latitude: 22.9167,longitude: 91.5)
+        let sylhet = CityModel(name: "Sylhet",latitude: 24.5,longitude: 91.6667)
+            
+        locationList = [dhaka, rajshahi, khulna, rangpur, barishal, chattogram, sylhet]
+    }
+}
+
+extension SearchCityViewController: UISearchBarDelegate {
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        searchData = []
+    
+        guard let searchInput = searchCity.text, let locationList = locationList else {
+            return
+        }
+        if searchInput.isEmpty {
+            searchData = locationList
+        }
+        else {
+            searchData = locationList.filter({
+                $0.name.lowercased().contains(searchText.lowercased())
```

Comment:
Do you have plan to integrate mapbox auto complete places? or it just filter the local city list>

Reply:
For now, I've created instances of the CityModel structure to filter from those instances. I'll do further updates in future.